### PR TITLE
Fixes emags wasting charges on un-emaggable stuff.

### PR DESCRIPTION
**Pull request by Ghommie**
:cl:
fix: Fixes emags wasting charges on un-emaggable & co stuff.
/:cl:

The limited uses feature for the emag is lame but what's more lame is how it still uses charges on stuff that can't be obviously emagged, already emagged objects, or stuff that require another action to be processed first (such as opening the panel) before emagging. This fixes it. Also made more or less all `emag_act()` either call the parent's or still send the component signal should they override.

### CODE REVIEW

File: code/game/machinery/limbgrower.dm [hunk 1]
```diff
@@ -224,3 +225,4 @@
 			stored_research.add_design(D)
 	to_chat(user, "<span class='warning'>A warning flashes onto the screen, stating that safety overrides have been deactivated!</span>")
 	obj_flags |= EMAGGED
+	return ..()
```
Comment (deathride58) [inline code/game/machinery/limbgrower.dm]: This calls the parent proc a second time

**Review by deathride58 — DISMISSED** [review]

**Comment by chaoko99** [comment]
Push this upstream.

**Comment by deathride58** [comment]
@chaoko99 TG doesn't have emag charges

**Review by silicons — DISMISSED** [review]

**Comment by Ghommie** [comment]
Merge this soon, thanks.

**Review by BlackMajor — APPROVED** [review]

**Pull request merged**